fem/nonlininteg.hpp

```diff
@@ -68,6 +69,38 @@ class NonlinearFormIntegrator
                                    ElementTransformation &Tr,
                                    const Vector &elfun);
 
+   // TODO: add support for other assembly levels (in addition to PA) and their
+   // actions.
+
+   // TODO: for mixed meshes the quadrature rules to be used by methods like
+   // Setup() can be given as a QuadratureSpace, e.g. using a new method:
+   // SetQuadratureSpace().
+
+   // TODO: the methods for the various assembly levels make sense even in the
+   // base class NonlinearFormIntegrator, except that not all assembly levels
+   // make sense for the action of the nonlinear operator (but they all make
+   // sense for its Jacobian).
```

Comment:
I think this TODO comment can be removed -- both here and in class `BilinearFormIntegrator` -- the TODO is addressed by this PR, at least to some extend.